turnkey-version

```diff
@@ -14,41 +14,91 @@
 # version.
 
 import sys
+import argparse
+from pathlib import Path
 
 import sysversion
 
-
-def usage(e=None):
-    if e:
-        print("error: " + str(e), file=sys.stderr)
-
-    print("Syntax: {} [rootfs]".format(sys.argv[0]), file=sys.stderr)
-    sys.exit(1)
+_TKL_VER = '/etc/turnkey_version'
 
 
 class TurnkeyVersionError(Exception):
     pass
 
 
 def main():
-    args = sys.argv[1:]
-
-    rootfs = "/"
-    if args:
-        if args[0] in ('-h', '--help'):
-            usage()
-
-        if len(args) != 1:
-            usage("incorrect number of arguments")
+    parser = argparse.ArgumentParser(
+            description='TurnKey Linux Version information')
+    # rootfs provides backwards compatability
+    parser.add_argument(
+            'rootfs', type=str, nargs="?",
+            help="Path to root filesystem. If not set, defaults to '/'.")
```

Comment:
rather than implicitly defaulting to `None` and checking later to set it to `'/'` you could register the argument with `default='/'`

Reply:
FWIW I did originally have that but I removed it. TBH, I'm not 100% sure why now... I think I went that way with the intention of supporting relative paths?